include/ppx/application.h

```diff
@@ -559,6 +564,18 @@ class Application
     // Initializes standard knobs
     void InitStandardKnobs();
 
+    // List gpus
+    void ListGPUs() const;
+
+    // Returns true to exit the loop
+    bool Mainloop();
```

Comment:
nit: this sounds to me like this will loop until it returns true/false.
What do you think about having 2 functions in the main loop, `Update()` and `Render()`, which splits the event/animation from the frame rendering?

Reply:
thanks for pointing this out, it makes sense. I made the change as suggested.